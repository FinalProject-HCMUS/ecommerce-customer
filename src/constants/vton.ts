export function getAvailableBodyParts() {
  return [
    'Upper-clothes',
    'Skirt',
    'Pants',
    'Dress',
    'Left-shoe',
    'Right-shoe',
    'Face',
    'Left-leg',
    'Right-leg',
    'Left-arm',
    'Right-arm',
  ];
}

export const FACE_PARTS = ['Face'];
export const BODY_PARTS = [
  'Upper-clothes',
  'Pants',
  'Skirt',
  'Dress',
  'Left-leg',
  'Right-leg',
  'Left-arm',
  'Right-arm',
];
