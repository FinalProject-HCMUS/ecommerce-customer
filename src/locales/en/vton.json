{
  "virtualTryOn": {
    "title": "Virtual Try-On",
    "uploadImage": "Upload your image",
    "tryOn": "Try On",
    "processing": "Processing...",
    "recoverFace": "Recover Face",
    "removeFace": "Remove Face",
    "selectService": "Select Service:",
    "fasterLessAccurate": "Faster - less accurate",
    "slowerMoreAccurate": "Slower - more accurate",
    "statusMessages": {
      "analyzingImage": "Analyzing image...",
      "processingFace": "Processing face data...",
      "generatingResult": "Generating try-on result...",
      "recoveringFace": "Recovering face..."
    },
    "placeholders": {
      "yourPhoto": "Your Photo",
      "facelessPhoto": "Faceless Photo",
      "selectGarment": "Select Garment",
      "result": "Result"
    }
  }
}