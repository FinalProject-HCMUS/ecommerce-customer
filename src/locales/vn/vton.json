{
  "virtualTryOn": {
    "title": "Thử đồ ảo",
    "uploadImage": "Tải lên hình ảnh của bạn",
    "tryOn": "Thử đồ",
    "processing": "Đang xử lý...",
    "recoverFace": "Khôi phục khuôn mặt",
    "removeFace": "Loại bỏ khuôn mặt",
    "selectService": "Chọn dịch vụ:",
    "fasterLessAccurate": "Nhanh hơn - ít chính xác",
    "slowerMoreAccurate": "Chậm hơn - chính xác hơn",
    "statusMessages": {
      "analyzingImage": "Đang phân tích hình ảnh...",
      "processingFace": "Đang xử lý dữ liệu khuôn mặt...",
      "generatingResult": "Đang tạo kết quả thử đồ...",
      "recoveringFace": "Đang khôi phục khuôn mặt..."
    },
    "placeholders": {
      "yourPhoto": "Ảnh của bạn",
      "facelessPhoto": "Ảnh không có khuôn mặt",
      "selectGarment": "Chọn trang phục",
      "result": "Kết quả"
    }
  }
}