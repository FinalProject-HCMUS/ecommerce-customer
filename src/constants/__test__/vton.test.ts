import { getAvailableBodyParts, FACE_PARTS, BODY_PARTS } from '../vton';

describe('vton constants', () => {
  test('getAvailableBodyParts should return all expected parts', () => {
    const expectedParts = [
      'Upper-clothes',
      'Skirt',
      'Pants',
      'Dress',
      'Left-shoe',
      'Right-shoe',
      'Face',
      'Left-leg',
      'Right-leg',
      'Left-arm',
      'Right-arm',
    ];
    expect(getAvailableBodyParts()).toEqual(expectedParts);
  });

  test('FACE_PARTS should contain only "Face"', () => {
    expect(FACE_PARTS).toEqual(['Face']);
  });

  test('BODY_PARTS should contain all expected body parts', () => {
    const expectedBodyParts = [
      'Upper-clothes',
      'Pants',
      'Skirt',
      'Dress',
      'Left-leg',
      'Right-leg',
      'Left-arm',
      'Right-arm',
    ];
    expect(BODY_PARTS).toEqual(expectedBodyParts);
  });

  test('should match snapshot', () => {
    expect({
      getAvailableBodyParts: getAvailableBodyParts(),
      FACE_PARTS,
      BODY_PARTS,
    }).toMatchSnapshot();
  });
});
