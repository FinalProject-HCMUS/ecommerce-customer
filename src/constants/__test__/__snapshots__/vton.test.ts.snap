// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vton constants should match snapshot 1`] = `
{
  "BODY_PARTS": [
    "Upper-clothes",
    "Pants",
    "Skirt",
    "Dress",
    "Left-leg",
    "Right-leg",
    "Left-arm",
    "Right-arm",
  ],
  "FACE_PARTS": [
    "Face",
  ],
  "getAvailableBodyParts": [
    "Upper-clothes",
    "Skirt",
    "Pants",
    "Dress",
    "Left-shoe",
    "Right-shoe",
    "Face",
    "Left-leg",
    "Right-leg",
    "Left-arm",
    "Right-arm",
  ],
}
`;
